lua/opencode/image_handler.lua

```diff
@@ -0,0 +1,156 @@
+--- Image pasting functionality from clipboard
+--- @see https://github.com/sst/opencode/blob/45180104fe84e2d0b9d29be0f9f8a5e52d18e102/packages/opencode/src/cli/cmd/tui/util/clipboard.ts
+local context = require('opencode.context')
+local state = require('opencode.state')
+
+local M = {}
+
+--- Check if a file exists and has content
+--- @param path string
+--- @return boolean
+local function is_valid_file(path)
+  return vim.fn.getfsize(path) > 0
+end
+
+--- Run shell command and return success
+--- @param cmd string
+--- @return boolean
+local function run_shell_cmd(cmd)
+  return vim.system({ 'sh', '-c', cmd }):wait().code == 0
+end
```

Comment:
purely stylistic, but you could have `run_shell_cmd` take an optional opts so that `save_base64` and others could use it as well.